app/Http/Controllers/AuthController.php

```diff
@@ -222,4 +224,75 @@ public function destroy($id)
     {
         //
     }
+
+    public function viewForgotPassword()
+    {
+        return view('auth.forgotPassword');
+    }
+
+    public function forgotPassword(Request $request)
+    {
+        $this->validate($request, [
+            'email' => 'required|email',
+        ]);
+        $user = User::where('email', $request->email)->firstOrFail();
+        if ($request->email == $user->email) {
+            $token = Str::random(64);
+
+            DB::table('password_resets')->insert([
+                'email' => $request->email,
+                'token' => $token,
+                'created_at' => Carbon::now('Asia/Ho_Chi_Minh')
+            ]);
+
+            Mail::send('email.forgetPassword', ['token' => $token], function ($message) use ($request) {
+                $message->to($request->email);
+                $message->subject('Reset Password');
+            });
+
+            return redirect('/')->with('mssg', 'We have e-mailed your password reset link! (Check your spam folder if you do not receive out email)');
+        } else {
+            return redirect('/')->with('mssg', 'cc');
+        }
+    }
+
+    public function viewResetPassword($token)
+    {
+        return view('auth.resetPassword', ['token' => $token]);
```

Comment:
Please add a check here
if DB doesn't store that token then render error screen
else render screen that display email